docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-andreperezmaselco.md

```diff
@@ -0,0 +1,203 @@
+# Google Summer of Code 2020 Report
+
+##### Project:
+##### [Fuzzing Graphics Shaders](https://summerofcode.withgoogle.com/projects/#6594591471435776)
+
+##### Student:
+##### [André Perez Maselco](https://github.com/andreperezmaselco)
+
+##### Organization:
+##### [Android Graphics Tools Team](https://summerofcode.withgoogle.com/organizations/6102438181863424/)
+
+##### Mentors:
+##### [Alastair Donaldson](https://github.com/afd)
+##### [Paul Thomson](https://github.com/paulthomson)
+
+## Abstract
+
+During the *Google Summer of Code 2020* program, I worked on the development of new semantics-preserving transformations, fuzzer passes and enhancements for the *spirv-fuzz* tool.
+Also, I developed [Amber Fuzz](https://github.com/andreperezmaselco/amber-fuzz), a simple graphics tool that takes an [Amber](https://github.com/google/amber) script as input and outputs a new script by fuzzing the reference shaders.
+
+## Introduction
+Graphics shaders are programs to run on a Graphics Processing Unit. Shaders can be written using the *OpenGL Shading Language* or *SPIR-V* (a simple binary intermediate language). Since shaders are programs, they need to be compiled into specific GPU instructions. A question that arises is what if the shader compiler has bugs? Some tools have been developed to deal with this problem.
+
+[SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools) is a collection of tools to work on *SPIR-V* modules.
+One of these tools is *spirv-fuzz*. It applies semantics-preserving transformations to a *SPIR-V* module. Theoretically, a semantics-preserving sentence should not change the behaviour of the variant code. If it has changed, then an error has occurred halfway, probably a compiler bug. In this way, the reference and variant shaders can be compared to check the differences. This approach, known as fuzzing, is part of technique called metamorphic testing. 
+
+The following is an example of transformation that replaces an `OpVectorTimesScalar` instruction with its mathematical definition. The vector components are extracted and multiplied by the scalar. Finally, the new vector is constructed using the multiplied components.
```

Comment:
As discussed, you could start a new section here. At least make sure you say something like "The following is an example of transformation that I implemented" or "that I proposed and implemented" (if true).

Reply:
Done.